src/tools/generate_insights.py

```diff
@@ -39,139 +63,206 @@ class Insight(BaseModel):
+
+Recharts data format:
+- Array of objects where each object represents one data point
+- Use simple, descriptive field names (e.g., 'date', 'value', 'category', 'count')
+- Ensure numeric values are actual numbers, not strings
+- For time data, use ISO date strings or simple date formats
 
 User's original query: {user_query}
 Raw data (in CSV format):
 {raw_data}
 
-Generate the insight based on the data and the user's query.
-""",
-        )
-    ]
-)
+Analyze this data and generate 1-2 compelling chart insights. Focus on the most important patterns that answer the user's query.
+
+For each insight:
+1. Choose the best chart type for the data pattern
+2. Transform data into Recharts format (array of objects with simple field names)
+3. Provide clear axis field names
+4. Write a compelling insight description
 
-insight_generation_chain = (
-    INSIGHT_GENERATION_PROMPT | sonnet.with_structured_output(Insight)
+Return 1 insight if the data is simple/focused, or 2 insights if there are multiple interesting patterns to explore.
+            """,
+        ),
+    ]
 )
 
 
-@tool("generate-insights")
+@tool
 def generate_insights(
     query: str,
-    aoi: str,
-    dataset: str,
+    aoi: str = "",
+    dataset: str = "",
     state: Annotated[Dict, InjectedState] | None = None,
     tool_call_id: Annotated[str, InjectedToolCallId] = None,
 ) -> Command:
     """
-    Analyzes raw data in the context of the user's query to generate a structured insight.
+    Analyzes raw data and generates 1-2 chart insights with Recharts-compatible data.
 
-    This tool takes the raw data pulled from a data source and uses an AI model
-    to generate a title, a short description, and a data table that represent
-    a key insight from the data.
+    This simplified tool combines insight planning and chart creation into a single step.
+    It analyzes the raw data and generates compelling visualizations that answer the user's query.
 
     Args:
-        query: The user's original query to provide context for the insight.
-        state: The current state of the agent, which must contain 'raw_data'.
+        query: The user's original query to provide context for insights.
+        aoi: Area of interest (optional, for context).
+        dataset: Dataset name (optional, for context).
     """
     logger.info("GENERATE-INSIGHTS-TOOL")
+    logger.debug(f"Generating insights for query: {query}")
 
-    raw_data = state.get("raw_data")
-    if not raw_data:  # raw_data is None or empty
-        logger.warning("No raw data found to generate insights from.")
+    if not state or "raw_data" not in state:
+        error_msg = "No raw data available in state. Please pull data first."
+        logger.error(error_msg)
         return Command(
             update={
                 "messages": [
                     ToolMessage(
-                        content="No raw data found to generate insights from.",
+                        content=error_msg,
```

Comment:
This is the pattern that we need to handle better, as it is hard to catch  on the frontend. Let's discuss how to flag these differntly.